lib/typeorm.providers.ts

```diff
@@ -11,12 +11,13 @@ export function createTypeOrmProviders(
   entities?: Function[],
   connection?: Connection | ConnectionOptions | string,
 ): Provider[] {
-  return (entities || []).map(entity => ({
+  return (entities || []).map((entity) => ({
     provide: getRepositoryToken(entity, connection),
     useFactory: (connection: Connection) => {
       if (
-        entity.prototype instanceof Repository ||
-        entity.prototype instanceof AbstractRepository
+        typeof Repository === 'function' &&
+        (entity.prototype instanceof Repository ||
+          entity.prototype instanceof AbstractRepository)
```

Comment:
Btw what about custom repositories in 0.3.0. Are they going to be deprecated/removed?

Reply:
You can extend a repository with custom methods by calling `extend()` on a repository instance:

```ts
repository.extend({
  customMethod() {
     // ...
  }
});
```

https://github.com/typeorm/typeorm/blob/b177e230a6fcb11f1eb71d4d431d0297436b7f6f/src/repository/Repository.ts#L380

Reply:
@kamilmysliwiec now (0.3.0) `@EntityRepository()` is deprecated. Here's how we should use instead: https://github.com/typeorm/typeorm/blob/master/docs/custom-repository.md